src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -1509,4 +1584,142 @@ public Snapshot getSnapshot(String clusterName, String snapshotName) {
+                getMetricsForClusterPrepStmt.bind(
+                    metricDomain, metricType, clusterName, timeBucket)));
+      }
+    }
+
+    for (ResultSetFuture future : futures) {
+      for (Row row : future.getUninterruptibly()) {
+        // Filtering on the timestamp lower bound since it's not filtered in cluster wide metrics requests
+        if (row.getTimestamp("ts").getTime() >= since) {
+          metrics.add(
+              GenericMetric.builder()
+                  .withClusterName(row.getString("cluster"))
+                  .withHost(row.getString("host"))
+                  .withMetricType(row.getString("metric_type"))
+                  .withMetricScope(row.getString("metric_scope"))
+                  .withMetricName(row.getString("metric_name"))
+                  .withMetricAttribute(row.getString("metric_attribute"))
+                  .withTs(new DateTime(row.getTimestamp("ts")))
+                  .withValue(row.getDouble("value"))
+                  .build());
+        }
+      }
+    }
+
+
+    return metrics;
+  }
+
+  @Override
+  public void storeMetric(GenericMetric metric) {
+    session.execute(
+        storeMetricsPrepStmt.bind(
+            metric.getClusterName(),
+            metric.getMetricDomain(),
+            metric.getMetricType(),
+            metric.getTs().toString(HOURLY_FORMATTER),
+            metric.getHost(),
+            metric.getMetricScope(),
+            metric.getMetricName(),
+            metric.getTs().toDate(),
+            metric.getMetricAttribute(),
+            metric.getValue()));
+  }
+
+  @Override
+  public void storeCompactions(String clusterName, String host, List<Compaction> activeCompactions)
+      throws JsonProcessingException {
+    session.executeAsync(
+        insertOperationsPrepStmt.bind(
+            clusterName,
+            OP_COMPACTION,
+            DateTime.now().toString(HOURLY_FORMATTER),
+            host,
+            DateTime.now().toDate(),
+            objectMapper.writeValueAsString(activeCompactions)));
+  }
+
+  @Override
+  public List<Compaction> listCompactions(String clusterName, String host)
+      throws IOException {
+    // cluster = ? AND type = ? and time_bucket = ? and host = ?
+    ResultSet compactions
+        = session.execute(
+            listOperationsForNodePrepStmt.bind(
+                clusterName, OP_COMPACTION, DateTime.now().toString(HOURLY_FORMATTER), host));
+    return compactions.isExhausted()
+        ? Collections.emptyList()
+        : parseJson(compactions.one().getString("data"), new TypeReference<List<Compaction>>(){});
+  }
+
+  @Override
+  public void storeStreams(String clusterName, String host, Set<CompositeData> activeStreams)
+      throws JsonProcessingException {
+    session.executeAsync(
+        insertOperationsPrepStmt.bind(
+            clusterName,
+            OP_STREAMING,
+            DateTime.now().toString(HOURLY_FORMATTER),
+            host,
+            DateTime.now().toDate(),
```

Comment:
this can be put into the CQL string as `now()`

Reply:
Well, I can't use the cql `now()` to fill in the hour that's used in the time bucket column, so I could end up with inconsistent times between the two fields.